src/resolve/resolved-dns-dnssec.c

```diff
@@ -625,6 +625,75 @@ static void dnssec_fix_rrset_ttl(
         rrsig->expiry = rrsig->rrsig.expiration * USEC_PER_SEC;
 }
 
+static int dnssec_rrset_serialize_sig(
+                DnsResourceRecord *rrsig,
+                const char *source,
+                DnsResourceRecord **list,
+                size_t list_len,
+                bool wildcard,
+                char **ret_sig_data,
+                size_t *ret_sig_size) {
+
+        _cleanup_free_ char *sig_data = NULL;
+        size_t sig_size = 0;
+        _cleanup_fclose_ FILE *f = NULL;
+        uint8_t wire_format_name[DNS_WIRE_FORMAT_HOSTNAME_MAX];
+        DnsResourceRecord *rr;
+        int r;
+
```

Comment:
I know we love to add them everywhere, but I don't think this makes much sense. This function is called from just one place, and we already verified that the variable is not null in the only caller, and used it. The only way that this could be null here would be if somebody edited the code and put an explicit NULL in the argument list… This is very unlikely and I don't think it makes sense to protect against that. In fact, this would a good scenario: we would get a very quick crash that would be trivial to interpret. A much more likely error is when either there's some memory corruption and the pointer stops being valid (but we don't know how to guard against that) or if somebody switches two pointers (and for some reason the type checks do not catch it). So I think the checks for NULL are just noise.

Reply:
They are often superfluous at the moment they are written, but I like that they clearly express what to expect for readers and future changes. Things we write here rarely stay untouched forever. It's very nice when going to modify a function, being able to immediately tell what I can expect to be valid and what needs to be checked, without having to trace back to all the callers.

Reply:
OK, I added a bunch.

Reply:
thanks!